app/models/user.rb

```diff
@@ -573,6 +575,19 @@ def bust_cache
     Users::BustCacheWorker.perform_async(id)
   end
 
+  def create_conditional_autovomits
+    return unless Settings::RateLimit.spam_trigger_terms.any? do |term|
+      Regexp.new(term.downcase).match?(name.downcase)
```

Comment:
We don't need to lowercase both strings, we just need a case-insensitive match. Ruby's regular expressions allow interpolation like strings, so this is easy to do:

```suggestion
      name.match?(/#{term}/i)
```

All the new specs still pass with this change.